src/firefly/js/drawingLayers/Catalog.js

```diff
@@ -0,0 +1,182 @@
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ */
+
+
+import {isEmpty} from 'lodash';
+import {primePlot} from '../visualize/PlotViewUtil.js';
+import {visRoot} from '../visualize/ImagePlotCntlr.js';
+import PointDataObj, {DrawSymbol} from '../visualize/draw/PointDataObj.js';
+import {makeDrawingDef, COLOR_PT_1, COLOR_PT_2, 
+    COLOR_PT_3, COLOR_PT_5, COLOR_PT_6} from '../visualize/draw/DrawingDef.js';
+import DrawLayer, {DataTypes,ColorChangeType} from '../visualize/draw/DrawLayer.js';
+import {makeFactoryDef} from '../visualize/draw/DrawLayerFactory.js';
+import DrawLayerCntlr from '../visualize/DrawLayerCntlr.js';
+import {MouseState} from '../visualize/VisMouseCntlr.js';
+import DrawOp from '../visualize/draw/DrawOp.js';
+import {makeWorldPt} from '../visualize/Point.js';
+import {dispatchTableHighlight} from '../tables/TablesCntlr.js';
+import {COLOR_HIGHLIGHTED_PT} from '../visualize/draw/DrawingDef.js';
+import {MetaConst} from '../data/MetaConst.js';
+
+const TYPE_ID= 'CATALOG_TYPE';
+
+const helpText=
+    `Click on point to highlight`;
+
+
+
+
+const factoryDef= makeFactoryDef(TYPE_ID,creator,getDrawData,getLayerChanges,null,null);
+export default {factoryDef, TYPE_ID}; // every draw layer must default export with factoryDef and TYPE_ID
+
+
+
+
+
+var createCnt= 0;
+const defColors= [COLOR_PT_1, COLOR_PT_2, COLOR_PT_3, COLOR_PT_5, COLOR_PT_6];
+
+
+function creator(initPayload) {
+    const {catalogId, tableData, tableMeta, columns, tableRequest, highlightedRow}= initPayload;
+    var drawingDef= makeDrawingDef();
+    drawingDef.symbol= DrawSymbol.SQUARE;
+
+    var pairs= {
+        [MouseState.DOWN.key]: highlightChange
+    };
+
+    drawingDef.color= tableMeta[MetaConst.DEFAULT_COLOR] || defColors[createCnt % defColors.length];
+
+    var options= {
+        hasPerPlotData:false,
+        isPointData:true,
+        canUserDelete: true,
+        canUseMouse:true,
+        canHighlight: true,
+        canSelect: true,
+        canFilter: true,
+        helpLine : helpText,
+        canUserChangeColor: ColorChangeType.STATIC
+    };
```

Comment:
One improvement we should add here (or to a different ticket) is that the user should be able to change the shape and size of the overlay symbol